sklearn/impute.py

```diff
@@ -600,6 +655,7 @@ def transform(self, X):
         X = check_array(X, accept_sparse=('csc', 'csr'),
                         force_all_finite=force_all_finite)
         _check_inputs_dtype(X, self.missing_values)
+        X = self._drop_constant_columns(X)
```

Comment:
This code is deceptive. It looks like it will use statistics from X to drop columns. Rather, perhaps, pass fully_missing_indices_ here.